### PR TITLE
Move fixchain to use client library

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: fixchain/fix_and_log_test.go [hunk 1]
```diff
@@ -192,15 +195,23 @@ var newFixAndLogTests = []fixAndLogTest{
 
 func TestNewFixAndLog(t *testing.T) {
 	// Test that expected chains are logged when adding a chain using QueueChain()
+	ctx := context.Background()
 	for i, test := range newFixAndLogTests {
 		seen := make([]bool, len(test.expLoggedChains))
 		errors := make(chan *FixError)
-		client := &http.Client{Transport: &testRoundTripper{t: t, test: &test, testIndex: i, seen: seen}}
-		fl := NewFixAndLog(1, 1, errors, client, client, test.url, newNilLimiter(), false)
+		c := http.Client{Transport: &testRoundTripper{t: t, test: &test, testIndex: i, seen: seen}}
```
Comment (taknira) [inline fixchain/fix_and_log_test.go]: Why remove the '&' to then only have to pass in c as '&c' everywhere it's used?  Why not keep c defined as a pointer?

File: fixchain/fix_and_log_test.go [hunk 2]
```diff
@@ -270,11 +281,17 @@ NextToFix:
 }
 
 func TestQueueAllCertsInChain(t *testing.T) {
+	ctx := context.Background()
 	for i, test := range fixAndLogQueueTests {
 		f := &Fixer{toFix: make(chan *toFix)}
+		c := http.Client{Transport: &testRoundTripper{}}
```
Comment (taknira) [inline fixchain/fix_and_log_test.go]: Similarly here, could just define c as a pointer to an http.Client then you don't have to take the address when passing to client.New below.

File: fixchain/fix_and_log_test.go [hunk 3]
```diff
@@ -305,11 +322,17 @@ func testFixAndLogQueueChain(t *testing.T, i int, test *fixAndLogTest, fl *FixAn
 }
 
 func TestFixAndLogQueueChain(t *testing.T) {
+	ctx := context.Background()
 	for i, test := range fixAndLogQueueTests {
 		f := &Fixer{toFix: make(chan *toFix)}
+		c := http.Client{Transport: &testRoundTripper{}}
```
Comment (taknira) [inline fixchain/fix_and_log_test.go]: And here, and in other tests files in this PR (so I don't keep writing 'and here') :)

File: fixchain/fix_and_log_test.go (hunk 1 above)
Comment (daviddrysdale) [reply fixchain/fix_and_log_test.go]: Done; I think comes from a residual C++-based fear of taking the address of a temporary.

File: fixchain/fix_and_log_test.go (hunk 3 above)
Comment (daviddrysdale) [reply fixchain/fix_and_log_test.go]: (I use 'passim' for this sentiment)

**Pull request merged**